src/DotNetLightning.Core/Utils/Primitives.fs

```diff
@@ -333,23 +335,24 @@ module Primitives =
         static member (*) (a: FeeRatePerKw, b: uint32) =
             (a.Value * b) |> FeeRatePerKw
     /// Block Hash
-    type BlockId = | BlockId of uint256 with
-        member x.Value = let (BlockId v) = x in v
+    [<Struct>]
```

Comment:
how about adding [<StructuralComparison;StructuralEquality>] instead of [<Struct>] ?

Reply:
Is there an advantage to that? My understanding is that `[<Struct>]` makes the type a thin wrapper around `uint256`, which seems to be what we want.

Reply:
@canndrew Define `thin wrapper`? There are many differences between classes and structs (e.g. allocated on stack instead of heap, AFAIR). But if what we care about here is only comparison, then I'd like to know if using `[<StructuralComparison;StructuralEquality>]` works, because if it does:
a) why not do this instead? the change would be more readable because it would declare intent by itself;
b) some cases I see which are now `[<Struct;CustomComparison;CustomEquality>]` may not need to change at all.

Reply:
To me "thin wrapper" means, at least in this case, that the outer layer of the value is allocated on the stack (so there aren't two pointer-indirections) and that the type forwards equality to the wrapped type.

I agree that `[<StructuralComparison;...>]` arguably expresses the intent for directly, however I had to add `[<Struct;...>]` to some of those types to get it to compile in the first place. I only added it to 3 more types when fixing the tests.

Reply:
Specifically, (since I just tried to build without `Struct` to see what the errors are), I get the error: *Only record, union, exception and struct types may be augmented with the 'ReferenceEquality', 'StructuralEquality' and 'StructuralComparison' attributes*.

This is the error I get if I, for instance, replace the `[<Struct;StructuralComparison;StructuralEquality>]` on `TxId` with just `[<StructuralComparison;StructuralEquality>]`.

Reply:
interesting... but in the case of this hunk:

```
-    [<CustomEquality;CustomComparison;StructuredFormatDisplay("{AsString}")>]
+    [<Struct;CustomEquality;CustomComparison;StructuredFormatDisplay("{AsString}")>]
```

why is Struct needed? if the equality and comparison are still custom! do tests pass if you remove this hunk from the PR?

Reply:
The error in this case is: *This type uses an invalid mix of the attributes 'NoEquality', 'ReferenceEquality', 'StructuralEquality', 'NoComparison' and 'StructuralComparison'*

Reply:
I don't understand, if that was the case master branch wouldn't compile at all, because it currently has `[<CustomEquality;CustomComparison;StructuredFormatDisplay("{AsString}")>]` before the changes from your PR.